src/services/call.ts

```diff
@@ -108,6 +109,30 @@ export class CallService {
         return this.myInstance;
     }
 
+    public static async makeOutgoingCall(user: User, props: any) {
```

Comment:
I don't like this code. Call service shouldn't have knowledge about component props. This service was architected as a reusable service between different projects. And also this service shouldn't have knowledge about Navigation.

You should just pass the method makeOutgoingCall from Call to Chat and handle isPending state in Chat component

Reply:
It isn't the only solution. Another solution is to create a service like fhir/saveFHIRResource, which accepts cursor to RemoteData.
But the first is simpler to implement.